service.rsspopup/service.py

```diff
@@ -0,0 +1,212 @@
+    xbmc.log('%s: %s' % (addonID, msg), level) 
+    
+    
+# Einlesen von Parametern, Notwendig fuer Reset der Twitter API
+def parameters_string_to_dict(parameters):
+	paramDict = {}
+	if parameters:
+		paramPairs = parameters[1:].split("&")
+		for paramsPair in paramPairs:
+			paramSplits = paramsPair.split('=')
+			if (len(paramSplits)) == 2:
+				paramDict[paramSplits[0]] = paramSplits[1]
+	return paramDict
+
+if not xbmcvfs.exists(temp):
+   xbmcvfs.mkdirs(temp)
+
+    
+def addrss():
+    dialog = xbmcgui.Dialog()
+    d = dialog.input("Url des Feeds", type=xbmcgui.INPUT_ALPHANUM)
+    filename       = xbmc.translatePath( os.path.join( temp, 'urlliste.txt') ).decode("utf-8")
+    if xbmcvfs.exists(filename) :
+      fp=open(filename,"r") 
+      content=fp.read()
+      fp.close()    
+      content=content+"\n"+d
+    else :
+       content=d
+    fp = open(filename, 'w')    
+    fp.write(content)
+    fp.close()    
+      
+def deleterss():
+   filename       = xbmc.translatePath( os.path.join( temp, 'urlliste.txt') ).decode("utf-8")
+   if xbmcvfs.exists(filename) :
+      fp=open(filename,"r") 
+      content=fp.read()
+      fp.close()          
+      liste=content.split("\n")
+      dialog = xbmcgui.Dialog()
+      nr=dialog.select(translation(30001), liste)
+      if nr<0:
+        return
+      liste.remove(liste[nr])
+      content="\n"
+      content=content.join(liste)
+      fp = open(filename, 'w')    
+      fp.write(content)
+      fp.close()    
+      
+   
+   
+# Soll Twitter Api Resetter Werden
+if len(sys.argv) > 1:
+    params = parameters_string_to_dict(sys.argv[2])
+    mode = urllib.unquote_plus(params.get('mode', ''))
+    if mode=="addrss":
+      addrss()
+    if mode=="deleterss":
+      deleterss()
+    
+
+
+
+def geturl(url):
+   req = urllib2.Request(url)
+   inhalt = urllib2.urlopen(req).read()   
+   return inhalt    
+          
+    
+if __name__ == '__main__':
+    cimg=""
+    xbmc.log("Twitter:  Starte Plugin")
+
+    schown=[]
+    monitor = xbmc.Monitor()   
+    
+    while not monitor.abortRequested():
+      titlelist=[]
+      cimglist=[]
+      greyoutlist=[]
+      lesezeitlist=[]
+      timelist=[] 
+      xbmc.log("Hole Umgebung")
+      bild=__addon__.getSetting("bild") 
+      lesezeit=__addon__.getSetting("lesezeit")
+      greyout=__addon__.getSetting("greyout")
+      xmessage=__addon__.getSetting("x-message")  
+      ymessage=__addon__.getSetting("y-message")  
+      hoehemessage=__addon__.getSetting("hoehe-message")  
+      breitemessage=__addon__.getSetting("breite-message")  
+      hoehebild=__addon__.getSetting("hoehe-bild")  
+      breitebild=__addon__.getSetting("breite-bild")  
+      font=__addon__.getSetting("font")  
+      fontcolor=__addon__.getSetting("fontcolor") 
+      filename       = xbmc.translatePath( os.path.join( temp, 'urlliste.txt') ).decode("utf-8")
+      gesamtliste=[]
+      if xbmcvfs.exists(filename) :
+        fp=open(filename,"r") 
```

Comment:
please use with statement

with open(filename,"r") as fp:
      content=fp.read()

Then drop the close execution